docs: give the README a full makeover with logo, screenshots, and badges (#784)
seamus-sloan/omnibus: README.md

<div align="center">

<img src="frontend/assets/omnibus-stoat.png" alt="Omnibus" width="120" />

# Omnibus

**The Plex / Jellyfin for your books.**

> A self-hosted ebook & audiobook library — read in the browser, listen anywhere, and browse a collection that actually looks like one.

Author: This part needs to resonate more with self-hosters.
```suggestion
A self-hosted ebook & audiobook library — read in the browser, listen anywhere, and browse a collection that belongs entirely to you.
```

Author: Applied your suggestion in 4daaaa0 — reframed to "belongs entirely to you."

Built with Rust ([Axum](https://github.com/tokio-rs/axum) + [Dioxus](https://dioxuslabs.com/)), SQLite, and a native iOS / Android shell.

> [![Rust CI](https://github.com/seamus-sloan/omnibus/actions/workflows/rust.yml/badge.svg?branch=main)](https://github.com/seamus-sloan/omnibus/actions/workflows/rust.yml)

Author: Instead of "Rust CI" can we call it `Clippy & Tests`?

Author: Done in 4daaaa0 — relabeled the badge to **Clippy & Tests** (switched to a shields.io workflow-status badge so the label is customizable; still tracks rust.yml on main).

> [![E2E](https://github.com/seamus-sloan/omnibus/actions/workflows/e2e.yml/badge.svg?branch=main)](https://github.com/seamus-sloan/omnibus/actions/workflows/e2e.yml)

Author: Let's call this playwright instead

Author: Done in 4daaaa0 — relabeled to **Playwright** via a shields.io workflow-status badge tracking e2e.yml.

[![CSS Lint](https://github.com/seamus-sloan/omnibus/actions/workflows/css-lint.yml/badge.svg?branch=main)](https://github.com/seamus-sloan/omnibus/actions/workflows/css-lint.yml)
[![Docker Hub](https://img.shields.io/docker/v/sesloan/omnibus?sort=semver&logo=docker&logoColor=white&label=docker%20hub&color=2496ED)](https://hub.docker.com/r/sesloan/omnibus/tags)
[![Image size](https://img.shields.io/docker/image-size/sesloan/omnibus?sort=semver&logo=docker&logoColor=white&label=image&color=2496ED)](https://hub.docker.com/r/sesloan/omnibus/tags)
[![Docker pulls](https://img.shields.io/docker/pulls/sesloan/omnibus?logo=docker&logoColor=white&color=2496ED)](https://hub.docker.com/r/sesloan/omnibus)

</div>

<div align="center">
<img src="docs/screenshots/library.png" alt="Omnibus library — cover grid" width="90%" />
</div>

> [!NOTE]
> **Status: early development.** Foundations and browse/discovery have shipped;
> reading/listening is in progress. The landing grid/table, EPUB reader,
> audiobook player, command-palette search, auth, and author/series/tag
> discovery are live. See the [roadmap](docs/roadmap/0-0-summary.md) for what's
> next.

> ## Features

Author: - Add `:book: **Device Uploads** - send your ebooks directly to your Kobo or Kindle device _(skip calibre!) _

Author: Added in 4daaaa0 — new **Send to your device** feature bullet (push ebooks to Kobo/Kindle, no Calibre round-trip).

- 📚 **A real library, not a file list** — cover-art grid or dense metadata table, plus smart & shared shelves that update themselves.
- 📖 **In-browser EPUB reader** — pick up where you left off, progress synced server-side.
- 🖍️ **Highlights & quote cards** — save passages and export a shareable quote card in a tap.
- 🎧 **Audiobook player** — HLS-transcoded streaming with a chapter map, resumable across devices.
- ⌘ **Command-palette search** — `⌘K` for full-text search across titles, authors, series, and tags.

> - 🧭 **Rich book pages** — ratings, a reading journal, series tracking, reading insights, and Send-to-Kindle / Kobo.

Author: ```suggestion
- 🧭 **Rich book pages** — your own ratings, reading journals, series tracking, and reading insights on your own database.
```

Author: Applied your suggestion in 4daaaa0.

- 🔗 **Discovery** — browse by author, series, and tag; "readers also enjoyed" suggestions via Hardcover.

> - 📱 **Native mobile app** — a thin iOS / Android shell over the same backend.

Author: ```suggestion
- 📱 **Native mobile app** — an iOS / Android shell over the same backend.
```

Author: Applied in 4daaaa0 — dropped "thin."

- 🐳 **Self-hosted, Jellyfin-style** — bind-mount your library, one `docker compose up`.

## Screenshots

<div align="center">

| Read & highlight | Listen |
|:---:|:---:|
| <img src="docs/screenshots/reader.png" alt="EPUB reader with quote-card composer" width="420" /> | <img src="docs/screenshots/player.png" alt="Audiobook player with chapter map" width="420" /> |
| **Search everything** | **Every book, in depth** |
| <img src="docs/screenshots/search.png" alt="Command-palette search" width="420" /> | <img src="docs/screenshots/discovery.png" alt="Book detail page" width="420" /> |

> 

Author: Should probably standardize the height of these...

Author: Fixed in 4daaaa0 — cropped the tall book-page shot to landscape and set a uniform `height="250"` on all four gallery images, so both rows are the same height now.

```

Full deployment reference — volumes, env vars, reverse-proxy/TLS, PUID/PGID, and
admin recovery — is in [docs/docker.md](docs/docker.md).

## Develop

Development uses [Nix](https://wiki.nixos.org/wiki/NixOS_Wiki) to pin every
dependency (Rust toolchain, SQLite, Node, mobile SDKs). **Always work inside the
dev shell.**

```bash
nix develop                 # slim default shell (spawns bash)
nix develop --command zsh   # …keeping your shell
direnv allow                # or auto-load via direnv (once)
```

The flake exposes purpose-built shells so daily cargo work doesn't pull in
Playwright + mobile + audit tooling. Opt into a heavier shell only when you need
it:

| Shell | When to use |
|---|---|
| `default` | Daily `cargo` / `clippy` / `test` / editor — what direnv auto-loads |
| `.#web` | `dx serve --platform web`, `just dev-up`, anything that bundles WASM |
| `.#e2e` | `npx playwright test` (the Chromium bundle lives here) |
| `.#mobile` | Android / iOS builds (Rust cross-targets + JDK + Android NDK detect) |
| `.#audit` | `cargo audit` / `cargo deny` |

`just serve`, `just serve-pc`, `just dev-up`, and `just dev-bounce` self-wrap in

> the right shell, so they run straight from `default`.

Author: The `just` recipes should be at the forefront of all of these. This is the simplest way for developers to get started.

Author: Restructured in 4daaaa0 — the `just` recipes now lead the Develop section (serve / serve-pc / dev-up / dev-bounce) with the Nix-shells table moved below as reference.

2. **Install the NDK** — **Tools → SDK Manager → SDK Tools → check "NDK (Side by side)" → Apply**
3. **Create an emulator** — **Tools → Device Manager → Create Virtual Device** (API 33+ recommended), then start it.
4. **Enter the mobile dev shell** — it auto-detects `ANDROID_HOME` / `ANDROID_NDK_HOME`:

   ```bash
   nix develop .#mobile --command zsh
   echo $ANDROID_HOME       # e.g. /Users/<you>/Library/Android/sdk
   echo $ANDROID_NDK_HOME   # e.g. …/sdk/ndk/28.x.x
   ```

   If they're empty, set them manually:

   ```bash
   export ANDROID_HOME=$HOME/Library/Android/sdk
   export ANDROID_NDK_HOME=$ANDROID_HOME/ndk/$(ls $ANDROID_HOME/ndk | tail -1)
   ```

5. If the app can't reach the server after boot, run `adb reverse tcp:3000 tcp:3000`.

## Tests

```bash
just test    # full matrix: db + server + frontend(--features server) + shared
just lint    # cargo fmt --check + clippy across the crate/feature matrix + stylelint
just check   # lint then test

# Web E2E (Playwright — server must be running; Chromium comes from Nix)
cd ui_tests/playwright
npm install                 # first time only; do NOT run `npx playwright install`
npx playwright test
```

> *Mobile UI tests will land later using [Maestro](https://github.com/mobile-dev-inc/maestro).*

Author: ```suggestion
*Mobile UI tests will land later using [MobileWright](https://mobilewright.dev/).*
```

Author: Applied in 4daaaa0 — swapped Maestro for MobileWright.
